### ISSUE TITLE
XHRConnection throws error on HTTP 400 responses

### ISSUE THREAD

**Issue by danielgerlag** [angular/angular#6490]
An HTTP 400 response is a valid response and should not invoke the observables error handler, it should still call the next handler.  The actual communication with the server did not fail, It is not an abnormal exception.

**Comment by zoechi** [comment]
I guess "new" should actually be "not"?

**Comment by danielgerlag** [comment]
Yes, sorry typo

**Comment by zoechi** [comment]
I guess this is a follow-up to #5130

**Comment by jeffbcross** [comment]
@danielgerlag can you specifically define `valid response`? 

Angular's HTTP is an application-level library with its design skewed more toward the application-layer than network-layer. As far as applications are concerned, anything other than 200 is usually considered an error. Apps can choose if they want to treat certain errors in certain cases from certain backends as valid responses, and can presently do so by writing services around Http. @robwormald mentioned that he's going to make the acceptable status code range configurable through DI, which would provide another option to solving this problem.

**Comment by dsebastien** [comment]
I'm hopping on this discussion as I'm facing this right now.

I've created a small RESTful back-end that returns various HTTP status codes depending on the operations and their results. Of course each and every back-end can design their services as they wish and either make extensive or limited use of the HTTP status codes.

On the client side, I've created a service to interact with my API and I was hoping to do something like: 
```
class ApiService {
   ...
   register:Observable<RegistrationResult>(registrationDetails:RegistrationDetails) {
       ...
       return http.post(...)
          .map(res:Response) {
             // check the response code and construct a RegistrationResult object accordingly
          };
    }
```

With the current behavior of Angular's Http utility, I'm not sure that my `map` function will be called in all cases (only when 200 OK?) and if not then I'll probably be forced to create another Observable to return to my service's clients. Of course it works, but it adds unnecessary boilerplate.

Being able to control this default behavior would indeed be nice to have
